subsys/debug/coredump/coredump_backend_flash_partition.c

```diff
@@ -273,7 +274,97 @@ static int process_stored_dump(data_read_cb_t cb, void *cb_arg)
 }
 
 /**
- * @brief Erase the stored coredump from flash partition.
+ * @brief Get the stored coredump in flash partition.
+ *
+ * This reads the stored coredump data and copies the raw data
+ * to the destination buffer.
+ *
+ * If the destination buffer is NULL, the offset and length are
+ * ignored and the entire dump size is returned.
+ *
+ * @param off offset of partition to begin reading
+ * @param dst buffer to read data into (can be NULL)
+ * @param len number of bytes to read
+ * @return dump size if successful; 0 if stored coredump is not found
+ *         or is not valid; error otherwise
+ */
+static int get_stored_dump(off_t off, uint8_t *dst, size_t len)
+{
+	int ret;
+	struct flash_hdr_t hdr;
+
+	ret = partition_open();
+	if (ret != 0) {
+		goto out;
+	}
+
+	/* Read header */
+	ret = data_read(0, (uint8_t *)&hdr, sizeof(hdr), NULL, NULL);
+	if (ret != 0) {
+		goto out;
+	}
+
+	/* Verify header signature */
+	if ((hdr.id[0] != 'C') && (hdr.id[1] != 'D')) {
+		ret = 0;
+		goto out;
+	}
+
+	/* Error encountered while dumping, so non-existent */
+	if (hdr.error != 0) {
+		ret = 0;
+		goto out;
+	}
+
+	/* Return the dump size if no destionation buffer available */
+	if (!dst) {
+		ret = (int)hdr.size;
+		goto out;
+	}
+
+	/* Offset larger than dump size */
+	if (off >= hdr.size) {
+		ret = 0;
+		goto out;
+	}
+
+	ret = data_read(off, dst, len, NULL, NULL);
```

Comment:
I am assuming the `off` here is used to do piece-wise copying. So you will need to account for the header size, and start copying from the first valid coredump data.

Reply:
Fixed